### PR TITLE
add starter redux bits

### PR DESCRIPTION
**Pull request by brendansudol**
This adds `redux` and related packages (`redux-logger`, `react-redux`), and initializes a file structure and example workflow for managing state (via a hello-world-esque `Counter` example).

Open JS Console to see info re: previous / next states as actions are dispatched.

### CODE REVIEW

File: web/src/app.js [hunk 1]
```diff
@@ -1,9 +1,18 @@
...
 
-ReactDOM.render(
-  <HelloWorld name="friendly visitor" />,
+const middleware = [createLogger()];
+const store = createStore(reducer, applyMiddleware(...middleware));
+
+render(
+  <Provider store={store}>
+    <App />
+  </Provider>,
```
Comment (mgwalker) [inline web/src/app.js]: 🙌  Thank you for moving the app bootstrap component into its own thing!

File: .codeclimate.yml [hunk 2]
```diff
@@ -1,4 +1,6 @@
 engines:
+  duplication:
+    enabled: false
```
Comment (mgwalker) [inline .codeclimate.yml]: I wish CodeClimate let you specific which files to apply this to.  It's a good idea, but tests are going to be ***FULL*** of duplication, intentionally.

File: web/src/containers/App.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+import Counter from '../components/Counter';
+import { decrement, increment } from '../actions';
+
+const App = ({ dispatch, total }) => (
+  <div>
+    <Counter
+      total={total}
+      onIncrement={() => dispatch(increment())}
+      onDecrement={() => dispatch(decrement())}
```
Comment (mgwalker) [inline web/src/containers/App.js]: I'm down with this, especially for something so simple.  How do you feel about the `mapDispatchToProps` approach for more complicated dispatch events?

File: .codeclimate.yml (hunk 2 above)
Comment (brendansudol) [reply .codeclimate.yml]: yep, i totally agree -- i really wish we could keep this on and ignore it for test files...

File: web/src/containers/App.js (hunk 3 above)
Comment (brendansudol) [reply web/src/containers/App.js]: yep, def prefer `mapDispatchToProps` (as well as `bindActionCreators`) once we start going for realsies

**Comment by brendansudol** [comment]
Thx @mgwalker, and agreed about chore issue around test coverage!

**Pull request merged**